iron-form.html

```diff
@@ -184,7 +184,7 @@
           // we won't be able to look into their shadowRoots for submittables.
           // We wait a tick and check again for any missing submittable default
           // values.
-          this.async(this._saveInitialValues.bind(this));
```

Comment:
I wonder if we should rather wait for the next render here with `Polymer.RenderStatus.afterNextRender` 🤔

Reply:
So this code path (in master) handles the case where the `<form>` is already distributed on attached, but its contents must still be upgraded.
My understanding of the issue is that the content is already upgraded but not yet _stamped_ (e.g. with `dom-repeat`), and we don't have tests for this.
Re `dom-repeat`, it can be forced to stamp with `Polymer.dom.flush()`, and I'm assuming `<vaadin-grid>` works the same way. Or, we can wait the same time it takes for it to stamp - is it after a RAF or a debounce?

Reply:
So I verified the issue with `dom-repeat`, see https://github.com/PolymerElements/iron-form/issues/261#issuecomment-375433181. It happens with native Shadow DOM only. This can be fixed by forcing distribution with `Polymer.dom.flush()` (which will execute the debounced rendering of dom-repeats). 

`dom-repeat` debounces the rendering, see https://github.com/Polymer/polymer/blob/0657de02ae8807b41ec909a742b4826378361430/lib/elements/dom-repeat.html#L501-L506

To solve the issue, we either increase the ms to 1 (as currently done) or we invoke `Polymer.dom.flush()` right before the async.

The only consequence I foresee for increasing the time to 1ms is that it will break tests already setup to wait 0ms, but won't break the functionality. @notwaldorf we should probably run tests for our internal projects to see if this is fine